docs/source/tutorial_optimization.rst

```diff
@@ -0,0 +1,891 @@
+                print('  absolute error w.r.t. reference solution: {:.2e}'.format(np.linalg.norm(result.x-reference_mu)))
+            print('  num iterations:     {}'.format(result.nit))
+            print('  num function calls: {}'.format(data['num_evals']))
+            print('  time: {:.5f} seconds'.format(data['time']))
+            if 'offline_time' in data:
+                print('  offline time: {:.5f} seconds'.format(data['offline_time']))
+            if 'enrichments' in data:
+                print('  model enrichments: {}'.format(data['enrichments']))
+        print('')
+
+Optimizing with the FOM using finite differences
+------------------------------------------------
+
+There exist plenty optimization methods and this tutorial is not meant
+to discuss the design and implementation of optimization methods. We
+simply use the ``minimize`` function from ``scipy.optimize`` and use the
+builtin ``L-BFGS-B`` routine which is a quasi-Newton method that can
+also handle a constrained parameter space.
+
+It is optional to give an expression for the gradient of the objective
+functional to the ``minimize`` function. In case no gradient is given,
+``minimize`` just approximates the gradient with finite differences.
+This is not recommended because the gradient is inexact and the
+computation of finite difference requires even more evalutations of the
+primal equation. We anyway start with this approach.
+
+.. jupyter-execute::
+
+    from functools import partial
+    from scipy.optimize import minimize
+    
+    tic = perf_counter()
+    fom_result = minimize(partial(record_results, fom_objective_functional, reference_minimization_data),
+                          initial_guess.to_numpy(),
+                          method='L-BFGS-B', jac=False,
+                          bounds=(ranges, ranges),
+                          options={'ftol': 1e-15})
+    reference_minimization_data['time'] = perf_counter()-tic
+    reference_mu = fom_result.x
+
+
+
+.. jupyter-execute::
+
+    report(fom_result, reference_minimization_data)
+
+
+Taking a look at the result, we see that the optimizer needs :math:`9`
+iterations to converge, but actually needs :math:`48` evaluations of the
+full order model. Obiously, this is related to the computation of the
+finite differences. We can visualize the optimization path by plotting
+the chosen points during the minimization.
+
+.. jupyter-execute::
+
+    reference_plot = plot_3d_surface(fom_objective_functional, XX, YY, alpha=0.5)
+    
+    for mu in reference_minimization_data['evaluation_points']:
+        addplot_xy_point_as_bar(reference_plot, mu[0], mu[1])
+
+
+
+
+Optimizing with the ROM using finite differences
+------------------------------------------------
+
+We can use a standard RB method to build a surrogate model for the FOM.
+As a result, the solution of the primal equation is no longer expensive
+and the optimization method has the chance to converge faster. For this,
```

Comment:
You mean "faster" in the sense that each solution to the primal problem is faster, or do you mean that less optimization steps are required for convergence?